modules/apps/headless/headless-builder/headless-builder-impl/src/main/resources/com/liferay/headless/builder/internal/batch/headless-builder.batch-engine-data.json

```diff
@@ -169,6 +169,77 @@
 					"system": false,
 					"type": "String"
 				},
+				{
+					"DBType": "String",
+					"businessType": "Picklist",
+					"externalReferenceCode": "PATH PARAMETER",
+					"indexed": true,
+					"indexedAsKeyword": false,
+					"label": {
+						"en_US": "Path Parameter"
+					},
+					"listTypeDefinitionExternalReferenceCode": "PATH_PARAMETER_PICKLIST",
+					"name": "pathParameter",
+					"objectFieldSettings": [
+						{
+							"name": "stateFlow",
+							"value": {
+								"objectStates": [
+									{
+										"key": "ID",
+										"objectStateTransitions": [
+										]
+									}
+								]
+							}
+						}
+					],
+					"required": false,
+					"system": false,
+					"type": "String"
+				},
+				{
+					"DBType": "String",
+					"businessType": "Picklist",
+					"defaultValue": "collection",
+					"externalReferenceCode": "RETRIEVE_TYPE",
+					"indexed": true,
+					"indexedAsKeyword": false,
+					"label": {
+						"en_US": "Retrieve Type"
+					},
+					"listTypeDefinitionExternalReferenceCode": "RETRIEVE_TYPE_PICKLIST",
+					"name": "retrieveType",
+					"objectFieldSettings": [
+						{
+							"name": "stateFlow",
+							"value": {
+								"objectStates": [
+									{
+										"key": "collection",
+										"objectStateTransitions": [
+											{
+												"key": "individual"
```

Comment:
Use the same naming as the UI (single element).

Reply:
There several restrictions when defining a PickList using the json. First of all I try to use the `single_element` key but, a key can not have blank spaces and special characters. Another idea that I have is putting the `name` property but it seems not working, it uses the key as a name too. Another idea is to define `singleElement`, and when we get the retrieve type, parse it as a enum of: 

```java
public enum RetrieveType{
    COLLECTION, SINGLE_ELEMENT
}
```
What do you think @4lejandrito?

Reply:
singleElement seems ok.